powertools-metrics/src/main/java/software/amazon/lambda/powertools/metrics/Metrics.java

```diff
@@ -183,6 +199,9 @@ default void captureColdStartMetric() {
      * @param namespace  the namespace for the metric
      */
     default void flushSingleMetric(String name, double value, MetricUnit unit, String namespace) {
-        flushSingleMetric(name, value, unit, namespace, null);
+        flushMetrics(metrics -> {
+            metrics.setNamespace(namespace);
+            metrics.addMetric(name, value, unit);
+        });
     }
```

Comment:
Following the discussion in the issue: All runtimes inherit the namespace by default if it is not overwritten. Let's add the method overload you suggested in your initial feature request here (without namespace param). We should also add a unit test for this.

```java
    default void flushSingleMetric(String name, double value, MetricUnit unit) {
        flushSingleMetric(name, value, unit, namespace, null);
        flushMetrics(metrics -> {
            metrics.addMetric(name, value, unit);
        });
```

Reply:
While this might make sense, a couple of thoughts
1. Not sure what value this would provide over `Metrics.addMetric` (I understand it's a different flush, but end result is that the same metric is emitted, reusing namespace, dimensions, metadata)
2. I sincerely believe that all `flushSingleMetric` should be marked as deprecated, to help reduce the api surface area as `flushMetrics` would already enable all that's supported by these overloads

Reply:
Interesting thought. Let's not add the additional overload for now. I'll think about deprecating these methods in a future v3 release. Let's not decide this in the scope of this PR.